### PR TITLE
Create utilities and governance contract folders

### PR DESCRIPTION
**Pull request by geoff-vball**
## Why this should be merged
closes https://github.com/ava-labs/teleporter/issues/410

## How this works
Reorganized directory structure.

## How this was tested
Existing tests

## How is this documented
Edited existing documentation.

### CODE REVIEW

File: contracts/src/teleporter/ReceiptQueue.sol
Comment (cam-schultz) [inline contracts/src/teleporter/ReceiptQueue.sol:1]: While we're here, what do you think of moving this file, `RentrancyGuards.sol`, and `SafeERC20TransferFrom.sol` to a new folder `contracts/src/utilities/utils`?

**Review by cam-schultz — COMMENTED** [review]
One comment about adding an additional generic `utilities` folder, but otherwise LGTM. Also, I know that lowercasing the directory names was mentioned in the issue, but did we come to a conclusion on that? If we decide to move forward with that, we should do the same in `avalanche-interchain-token-transfer`.

**Comment by iansuvak** [comment]
> Also, I know that lowercasing the directory names was mentioned in the issue, but did we come to a conclusion on that? If we decide to move forward with that, we should do the same in `avalanche-interchain-token-transfer`.

It was brought up in #404 and I think it makes sense overall. FWIW OZ uses lower cased directory names and ProperCase contract file names. I agree that if  we did it here we should do the same in `avalanche-interchain-token-transfer` repo as well. 

Overall LGTM. I just have one minor question below.

File: contracts/src/utilities/validators/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-## Utility Contracts
```
Comment (iansuvak) [inline contracts/src/utilities/validators/README.md:1]: I agree that it makes sense to separate different logical groups in sub-directories but I'm not sure about the `validators` name. Does anyone have a better idea? Maybe `governance`? I think we are more likely to have another governance type contract that would make sense to group this with than a validator related one?

File: contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol
Comment (iansuvak) [inline contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol:1]: In the original directory mockup on the #404 we agreed to keep Upgradeable files inside the `teleporter` folder.  Apologies for not copying it properly into the issue. I'm not opposed to keeping them here though

Comment (geoff-vball) [reply contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol:1]: Hmm, I do think it makes sense to keep `TeleporterRegistry` and `TeleporterUpgradeable` in the same spot. Would love to hear other opinions if anyone has one :)

File: contracts/src/teleporter/ReceiptQueue.sol
Comment (geoff-vball) [reply contracts/src/teleporter/ReceiptQueue.sol:1]: Sounds good to me!

File: contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol
Comment (cam-schultz) [reply contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol:1]: I agree we should keep them together, especially if we rename `TeleporterUpgradeable` as discussed [here](https://github.com/ava-labs/teleporter/pull/397#discussion_r1671342631)

File: contracts/src/utilities/validators/README.md (hunk 1 above)
Comment (minghinmatthewlam) [reply contracts/src/utilities/validators/README.md:1]: i like governance

**Review by minghinmatthewlam — COMMENTED** [review]
I overall still feel it's a bit strange to classify `TeleporterRegistry` and the other `upgrades` contracts under `utilities`, and the same for `governance` files. Thinking if those can sibling directories to `teleporter` or under `teleporter`. That way we can also remove the need to have another `utils` folder under `utilities`.

**Review by bernard-avalabs — COMMENTED** [review]
Generally looks good to me. I agree with some of the name change suggestions made by others, such as validators->governance.

**Comment by geoff-vball** [comment]
I'd be okay having `governance` and `upgrades` as top level directories, then we don't have to have `utilities/utils`.

**Comment by geoff-vball** [comment]
@iansuvak I think I like the lowercase names. If we go ahead with https://github.com/ava-labs/teleporter/issues/419, then we can lowercase the directories in ICTT when we move them over.

**Comment by iansuvak** [comment]
> @iansuvak I think I like the lowercase names. If we go ahead with #419, then we can lowercase the directories in ICTT when we move them over.

I agree and I think it makes sense to do it together with #419 implementation instead of a separate issue.

File: contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol
Comment (michaelkaplan13) [reply contracts/src/utilities/upgrades/TeleporterOwnerUpgradeable.sol:1]: I was originally thinking we'd put the registry contract under `utilities` but keep the `TeleporterUpgradeable`, etc under `Teleporter`. Names aside, the registry is a standalone contract that isn't inherently tied to Teleporter itself (could be used a general account versioning mechanism controlled by the validator set). The `TeleporterUpgradeable` contracts OTOH are abstract contracts that are tied to Teleporter in the sense that they import Teleporter interfaces and provide helpers specifically for sending and receiving Teleporter messages (by leveraging a registry instance).

File: contracts/src/governance/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-## Utility Contracts
+# ValidatorSetSig Contract
```
Comment (iansuvak) [inline contracts/src/governance/README.md:1]:
```suggestion
# Governance Contracts
```

nit: Can we change this back and `## Overview` below to `## ValidatorSetSig Contract` now that we have a `governance` folder?

**Comment by minghinmatthewlam** [comment]
> I'd be okay having `governance` and `upgrades` as top level directories, then we don't have to have `utilities/utils`.

Definitely agree for `governance`. For `upgrades` I think the standard is to use it only in context of upgradeability in contracts, and think we should stick with that, so prefer `upgrades`-> `registry`. Then because the `registry` directory and all contracts under are tied to `teleporter`, i'd prefer to leave under `teleporter` folder but with the rename.

**Review by cam-schultz — APPROVED** [review]

**Review by iansuvak — APPROVED** [review]
LGTM!

**Review by minghinmatthewlam — APPROVED** [review]
LGTM

**Pull request merged**